RosNet/DataModel/RosBag.cs

```diff
@@ -43,6 +45,53 @@ public override string ToString()
             return s;
         }
 
+        /// <summary>
+        /// Makes a dictionary of all the topics and fieldnames in rosbag
+        /// </summary>
+        /// <returns>Dictionary with topics and corresponding fieldnames</returns>
+        public Dictionary<string, List<string>> GetConnectionFields()
+        {
+            var fieldsByTopic = new Dictionary<string, List<string>>();
+            foreach (Connection conn in Connections.Values)
+            {
+                fieldsByTopic.Add(conn.Topic, new List<string>());
+                foreach (FieldValue field in conn.MessageDefinition)
+                {
+                    fieldsByTopic[conn.Topic].Add(field.Name);
+                }
+            }
+            return fieldsByTopic;
+        }
 
+        /// <summary>
+        /// Makes a dictionary of timestamps and fieldvalues of field with fieldname fieldName
+        /// </summary>
+        /// <returns>Dictionary with time and corresponding fieldvalue</returns>
+        public Dictionary<Time, FieldValue> GetTimeSeries(string topic, string fieldName)
```

Comment:
Why have you chosen to use a dictionary?

Reply:
We thought it would be nice if you were able to query for data at specific times, and thought dict was a good way to get that functionality. And at the same time, you can still easily iterate over it for the full data series

Reply:
We changed this to a list of tuples now